compiler/noirc_evaluator/src/ssa/ir/dfg.rs

```diff
@@ -21,6 +22,75 @@ use serde::{Deserialize, Serialize};
 use serde_with::serde_as;
 use serde_with::DisplayFromStr;
 
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub(crate) struct CallStackId(u32);
+
+impl CallStackId {
+    pub(crate) fn root() -> Self {
+        Self::new(0)
+    }
+
+    fn new(id: usize) -> Self {
+        Self(id as u32)
+    }
+
+    pub(crate) fn index(&self) -> usize {
+        self.0 as usize
+    }
+
+    pub(crate) fn is_root(&self) -> bool {
+        self.0 == 0
+    }
+
+    // Retrieves a CallStack from a CallStackId
+    fn get_call_stack(&self, locations: &[LocationNode]) -> CallStack {
+        let mut call_stack = List::new();
+        let mut current_location = *self;
+        while let Some(parent) = locations[current_location.index()].parent {
+            call_stack.push_back(locations[current_location.index()].value);
```

Comment:
Just to confirm: if a location doesn't have a parent, its own value is not going to be part of the call stack either? Like I guess `root()` doesn't have a parent, and you don't expect `root.value` to be interesting?

Reply:
yes, root means an empty call stack.